src/main/java/com/ferreusveritas/dynamictrees/blocks/BlockBranch.java

```diff
@@ -436,22 +438,81 @@ public Species getSpeciesFromSignal(World world, MapSignal signal) {
 	}
 	
 	// Destroys all branches recursively not facing the branching direction with the root node
-	public int destroyTreeFromNode(World world, BlockPos pos) {//, float fortuneFactor) {
+	public int destroyBranchFromNode(World world, BlockPos pos, boolean wholeTree) {
 		MapSignal signal = analyse(world, pos, null, new MapSignal());// Analyze entire tree network to find root node
 		Species species = getSpeciesFromSignal(world, signal);//Get the species from the root node
 		NodeNetVolume volumeSum = new NodeNetVolume();
+		NodeDestroyer destroyer = new NodeDestroyer(species);
 		// Analyze only part of the tree beyond the break point and calculate it's volume
-		analyse(world, pos, signal.localRootDir, new MapSignal(volumeSum, new NodeDestroyer(species)));
+		analyse(world, pos, wholeTree ? null : signal.localRootDir, new MapSignal(volumeSum, destroyer));
+		destroyLeaves(world, species, destroyer.getEnds());//Destroy all the leaves on the branch
 		return volumeSum.getVolume();// Drop an amount of wood calculated from the body of the tree network
 	}
+
+	/*
+	 * Attempt to destroy all of the leaves on the branch while leaving the other
+	 * leaves unharmed.
+	 */
+	protected void destroyLeaves(World world, Species species, List<BlockPos> endPoints) {
+
+		if(endPoints.isEmpty()) {
+			return;
+		}
 	
-	public int destroyEntireTree(World world, BlockPos pos) {
-		MapSignal signal = analyse(world, pos, null, new MapSignal());// Analyze entire tree network to find root node
-		Species species = getSpeciesFromSignal(world, signal);//Get the species from the root node
-		NodeNetVolume volumeSum = new NodeNetVolume();
-		// Analyze the entire tree and calculate it's volume
-		analyse(world, pos, null, new MapSignal(volumeSum, new NodeDestroyer(species)));
-		return volumeSum.getVolume();// Drop an amount of wood calculated from the body of the tree network
+		BlockPos firstPos = endPoints.get(0);
```

Comment:
Too many code in one method.
Devide to logical part. Instead of writing comments.
It will be looks better from clean code point of view.

Reply:
What do you mean? This is exactly how it's on the 1.12 branch, just backported to 1.7.10.

Reply:
You can't have code after a `return` statement.

Reply:
@SkyHawkB Obviously, but there is no code after a return statement, is there?

Reply:
Oh silly me, I misread the changes. Sorry.